doc/src/dump_image.rst

```diff
@@ -957,6 +959,20 @@ frequently.
 
 ----------
 
+.. versionadded:: TBD
+
+The *fsaa* keyword can be used with the dump image command to improve
+the image quality by enabling full scene anti-aliasing.  Internally the
+image is rendered at twice the width and height and then scaled down by
+computing the average of each 2x2 block of pixels to produce a single
+pixel in the final image at the original size. This produces images with
+smoother, less ragged edges.  The calculation of this effect can
+increase the cost of computing the image by roughly 4x or more,
+especially for large images and with large processor counts due to
```

Comment:
"increased communication cost" is not actually the issue here - it is that the last stage (shrink image with aliasing) is done serially on a single proc - so no parallelism and no communication.

So I think my sentence is better, but add a phrase to clarify:

This option will increase the cost of computing the image by 4x, or somewhat more for large images
on large processor counts due to the final anti-aliasing operation being performed on a single processor.

Reply:
I think you are overestimating the cost of the downscaling. The way it is done is very cache friendly and the really slow step is the part were the data is actually written to a file (and for JPEG and PNG there is also a compression step). So the communication cost is something that is significant because the Image::merge() function is collecting the image data into one buffer on rank 0. With a 4x larger image, that is more expensive. A more sophisticated implementation would do the downscaling **before** the "merge" step. But I have not looked into it. Also the downscaling could be easily made thread-parallel, but that would require creating a new buffer and it would be more memory bandwidth limited rather than CPU time limited, since the cost of computing the average of 4 numbers is negligible (the bicubic algorithm would be *must* more expensive since up to 16 pixels will have to be looked at and cubic polynomials evaluated.

Reply:
I'm thinking about it differently.  The cost of creating an image in parallel on P procs (separate from writing it out)
has calculation and communication costs.  To create an image 4x larger on the same P procs costs 4x more b/c the calculation and communication costs both go up by 4x.  4x more pixels to calculate and 4x more to communicate in each message.  The more-than-4x cost for ffsa is extra serial computation at the end.  But maybe you are right, that extra operation will never be that significant.

That said, I'm fine with the doc page saying:  "This option will increase the cost of computing the image by 4x or more."
I don't think SSAO is precisely 2x more expensive.

Reply:
I forgot that after each comm stage, the receiving procs need to merge 2 images.  Which requires them to non-scalably loop over all the pixels.  So that is extra computation cost which happens log2(P) times when the image size is 4x larger.  It is also likely similar cost to the one-time anti-alias loop over all pixels.  So I think saying "4x or more" is sufficient info for the user.